crates/next-core/src/server_rendered_source.rs

```diff
@@ -231,13 +237,24 @@ async fn create_server_rendered_source_for_directory(
                             // pageExtensions option from next.js
                             // defaults: https://github.com/vercel/next.js/blob/611e13f5159457fedf96d850845650616a1f75dd/packages/next/server/config-shared.ts#L499
                             "js" | "ts" | "jsx" | "tsx" => {
-                                let (dev_server_path, intermediate_output_path) =
+                                let (dev_server_path, intermediate_output_path, specificity) =
                                     if basename == "index" {
-                                        (server_path.join("index.html"), intermediate_output_path)
+                                        (
+                                            server_path.join("index.html"),
+                                            intermediate_output_path,
+                                            specificity,
+                                        )
+                                    } else if basename == "404" {
+                                        (
+                                            server_path.join("[...]"),
+                                            intermediate_output_path,
+                                            specificity,
+                                        )
                                     } else {
                                         (
                                             server_path.join(basename).join("index.html"),
                                             intermediate_output_path.join(basename),
+                                            specificity,
```

Comment:
It doesn't look like we ever change `specificity` here, so why rebind it?

Reply:
yeah, good point. We actually need to change specificity for the 404 route